File: FE/Seo-garden/투포인터_슬라이딩윈도우/Silver/3273_두수의합.swift
import Foundation

let n = Int(readLine()!)!
let a = readLine()!.split(separator: " ").map { Int($0)! }.sorted()
let sum = Int(readLine()!)!

var left = 0
var right = a.count - 1
var count = 0

while left < right {                //왼쪽 좌표와 오른쪽 좌표가 만날 때 까지
    if a[left] + a[right] == sum {  //sum 와 값이 동일하면
        count += 1
        left += 1                   //좌표 한칸씩 이동
        right -= 1
    } else if a[left] + a[right] < sum {
        left += 1
    } else if a[left] + a[right] > sum {
        right -= 1
    }
}

print(count)

File: BE/lhdmir/투 포인터/Silver/3273_두 수의 합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());

        int[] arr = new int[n];
        StringTokenizer st = new StringTokenizer(br.readLine());

        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(arr);

        int x = Integer.parseInt(br.readLine());

        int left = 0, right = n - 1;
        int sum;
        int count = 0;
        while (left < right) {
            sum = arr[left] + arr[right];

            if (sum > x) {
                right--;
            } else if (sum < x) {
                left++;
            } else {
                count++;
                left++;
                right--;
            }
        }

        System.out.println(count);
    }
}